app/models/collection.rb

```diff
@@ -19,8 +19,8 @@ class Collection < ActiveRecord::Base
   ## TODO: This potential has locking issues
   validates_with UniqueForOwnerValidator
 
-  can_by_role :destroy, :update, :update_links, :destroy_links,
-              roles: [ :owner, :collaborator ]
+  can_by_role :destroy, :update, :destroy_links, roles: [ :owner, :collaborator ]
+  can_by_role :update_links, roles: [ :owner, :collaborator, :contributor ]
   can_by_role :index, :show, public: true, roles: [ :owner, :collaborator, :viewer ]
```

Comment:
I figured since the only view restrictions on collections are public vs. private that it didn't matter, but it looking again it does seem cleaner to add it there.